upa/contracts/Deposits.sol

```diff
@@ -0,0 +1,283 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {EIP712} from "@openzeppelin/contracts/utils/cryptography/EIP712.sol";
+import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "./IUpaVerifier.sol";
+
+error NoPendingWithdrawal();
+error InsufficientNotice();
+error InsufficientBalance();
+error FailedToSendEther();
+error AlreadyClaimed();
+error AlreadyRefunded();
+error BadSignature();
+error UnauthorizedAggregatorAccount();
+error VerifiedBeforeExpiry();
+error NotPastExpiry();
+error UpaAddressIsZero();
+error AggregatorAddressIsZero();
+
+/// Contract that off-chain aggregators may use for handling fee deposits,
+/// payments, and claims.
+contract Deposits is EIP712 {
+    struct SubmitterAccount {
+        /// Remaining balance held by the Submitter
+        uint256 balance;
+        /// Total amount claimed by the aggregator so far
+        uint256 totalClaimed;
+        /// The block when a pending withdrawal was initiated. Zero if there
+        /// is no pending withdrawal.
+        uint256 pendingWithdrawalInitializedAtBlock;
```

Comment:
@dysfunctor  Should we make this `canWithdrawAtBlock`, so the owner can't extend the time window after a withdrawal has been initailized?

Reply:
(can be an issue for next version)

Reply:
#1106